### PR TITLE
Part of #20163 for file ui\components\app\modals\eth-sign-modal\eth-sign-modal.js

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation
This PR is a part of the issue #20163 Replace deprecated CheckBox component with Checkbox from the component-library for file
`ui\components\app\modals\eth-sign-modal\eth-sign-modal.js`

Also updated the `Box` component with `Box` component from the component-library.

### Before
image

### After
image

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
If you get time, kindly give some ideas to help me tackle the problem stated in the [comment](https://github.com/MetaMask/metamask-extension/pull/20537#issuecomment-1689481553) ?

**Review by georgewrmarshall — COMMENTED** [review]
Code looks good just need to update snapshots and I'll do a full review! Have commented in the thread you posted as well

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
As you asked in the [comment](https://github.com/MetaMask/metamask-extension/pull/20537#issuecomment-1694386109),
I am using Windows.

Thank you for helping me out.
Will definitely ask @HowardBraham for help.

Regards,
Harsh

**Comment by PrgrmrHarshShukla** [comment]
Hello there @HowardBraham ,
If possible, kindly share some ideas on how I can solve the following problem.

I am using a **Windows Operating System.**

I am not able to update snapshots using the command `yarn jest -u`
I am getting these errors continuously.
1.)
image
2.)
image
Since `JSDOM()` is being taken into use on line 52 in `test\helpers\setup-helper.js`, I am not able to make out why the second error is there.


Kindly look at this [comment](https://github.com/MetaMask/metamask-extension/pull/20537#issuecomment-1689481553) for some more context.

Regards,
Harsh

**Comment by HowardBraham** [comment]
@PrgrmrHarshShukla This is a Node version problem.  I bet you have Node v18.16.x.  I just reproduced it myself.
This issue was okay in v18.15.0, broke in v18.16.0, fixed in v18.17.0: nodejs/node/issues/47563

**Comment by PrgrmrHarshShukla** [comment]
You are absolutely right.
I actually have node `v18.16.1`.
Will move to `v18.15.0` or `v18.17.0`.

Thanks a lot for helping me out.

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
All the tests have passed for this PR.
Kindly look into this so that any remaining problems can be solved.

Regards,
Harsh

**Comment by PrgrmrHarshShukla** [comment]
> Hey @georgewrmarshall , All the tests have passed for this PR. Kindly look into this so that any remaining problems can be solved.
> 
> Regards, Harsh

Hey @georgewrmarshall , @garrettbear ,
Please look into this.

Regards,
Harsh

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Pushed some small fixes to attributes and props while I had the branch pulled. Thanks for your contribution @PrgrmrHarshShukla!

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
Probably we need to update the snapshots.
Shall I add a commit for that?

**Comment by PrgrmrHarshShukla** [comment]
Thanks a lot for doing all these changes to these PRs yourself. 🙏🏼
I am learning a lot through them and working hard to bring better PRs in the future.
I am trying my best to open PRs for only those issues I can solve really well and with minimum or no errors at all unlike the previous ones.
I hope, this way, I will improve with time and do not waste the precious time of reviewers.

**Comment by georgewrmarshall** [comment]
Hey @PrgrmrHarshShukla, no problem! I appreciate you taking on these tasks and helping us reduce tech debt and improve the UI consistency of the extension! And yes! Any snapshot test failures are mistakes on my side please go ahead and update whenever you see failing tests 🙏

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @PrgrmrHarshShukla

**Comment by PrgrmrHarshShukla** [comment]
Hey @garrettbear ,
Your help is probably needed here too.

**Review by garrettbear — APPROVED** [review]

File: ui/components/app/modals/eth-sign-modal/eth-sign-modal.js [hunk 1]
```diff
@@ -130,20 +129,16 @@ const EthSignModal = ({ hideModal }) => {
           alignItems={AlignItems.flexStart}
           gap={2}
         >
-          <CheckBox
-            id="eth-sign__checkbox"
+          <Checkbox
+            id="eth-sign-checkbox"
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/modals/eth-sign-modal/eth-sign-modal.js:133]: No reason I think this was a preconceived notion that we shouldn't use underscores in id names. Looks doesn't look like it's used anywhere other than here so it's safe to update

Screenshot 2024-01-23 at 8 48 39 PM

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**